storage-reporter/src/main/java/org/duracloud/mill/storagereporter/StorageReporter.java

```diff
@@ -0,0 +1,199 @@
+        List<AccountStorageReportResult> undersubscribedAccounts = new LinkedList<>();
+
+        Date now = new Date();
+        Calendar c = Calendar.getInstance();
+        c.add(Calendar.MONTH, -1);
+        Date lastMonth = c.getTime();
+        List<AccountInfo> accounts = accountRepo
+                .findByStatus(AccountStatus.ACTIVE);
+        Collections.sort(accounts, new Comparator<AccountInfo>() {
+            @Override
+            public int compare(AccountInfo o1, AccountInfo o2) {
+                return o1.getAcctName().compareTo(o2.getAcctName());
+            }
+        });
+
+        for (AccountInfo account : accounts) {
+            String accountId = account.getSubdomain();
+            LOGGER.info("processing {}", accountId);
+            // gather all storage providers for an account into a single list.
+            StorageProviderAccount primary = account
+                    .getPrimaryStorageProviderAccount();
+            Set<StorageProviderAccount> secondary = account
+                    .getSecondaryStorageProviderAccounts();
+            List<StorageProviderAccount> providers = new LinkedList<>(secondary);
+            providers.add(0, primary);
+
+            AccountStorageReportResult result = new AccountStorageReportResult(account);
+
+            // for each storage provider
+            for (StorageProviderAccount storageProviderAccount : providers) {
+                List<Object[]> stats = statsRepo
+                        .getByAccountIdAndStoreId(accountId,
+                                                  storageProviderAccount.getId()
+                                                          + "",
+                                                  lastMonth,
+                                                  now,
+                                                  JpaSpaceStatsRepo.INTERVAL_DAY);
+                long total = 0;
+                if (stats != null && stats.size() > 0) {
+                    total = ((BigDecimal) stats.get(stats.size() - 1)[3])
+                            .longValue();
+                }
+                result.addStorageProviderResult(storageProviderAccount, total);
+            }
+
+            if (result.isOversubscribed()) {
+                oversubscribedAccounts.add(result);
+            } else {
+                undersubscribedAccounts.add(result);
+            }
+
+        }
+
+        // build subject
+        String subject = MessageFormat.format(
+                                              "Storage Report:  {0} Oversubscribed account(s)",
+                                              oversubscribedAccounts.size());
+
+        StringBuilder body = new StringBuilder();
+
+        if (oversubscribedAccounts.size() == 0) {
+            body.append("Presently there are no oversubscribed accounts.\n\n");
+        } else {
+            body.append("Oversubscribed Accounts: \n\n");
```

Comment:
This may be easier to understand if "Oversubscribed Accounts" is changed to "Accounts OVER their subscribed storage limit"